### PR TITLE
Added Sparkfun MicroMod RP2040

### PR DESCRIPTION
**Pull request by stonehippo**
Added board definition for the Sparkfun MicroMod RP2040.

Mostly followed the format for pins from the MicroMod nRF52840 board definition, as MicroMod tries to standardize pins across microcontrollers.

The MicroMod differs a bit from other SparkFun RP2040 boards in how it handles some pins (for example, it treats some pins as dedicated PWM or digital). To keep to the spirit of how the pins are defined in the datasheet, I have not mapped pins to a general `Dx` alias as was done with the SparkFun Pro Micro RP2040 and SparkFun Thing Plus RP2040 boards. The `Gx` format used by MicroMod is used for generic GPIO pins.

### CODE REVIEW

**Comment by ladyada** [comment]
hihi please run pylint/black or pre-commit on the files so CI passes and i'll take a look!

**Review by ladyada — DISMISSED** [review]

**Comment by stonehippo** [comment]
@ladyada all CI checks passing now! Please review when you have a chance.

**Comment by ladyada** [comment]
thanks, did Sparkfun give you taht PID to use? if not please ask them for a unique one so we can identify this board in circuitpython!

**Comment by stonehippo** [comment]
I’m checking in their forum, will confirm when I hear back.

https://forum.sparkfun.com/viewtopic.php?f=182&t=55251

**Comment by tannewt** [comment]
0x0024 is correct. I got them to assign USB PIDs up front and documented them here: https://github.com/adafruit/circuitpython/issues/4042

**Comment by stonehippo** [comment]
Thanks @tannewt! (I used the PID from that post, but wasn't sure how official it was). @ladyada anything else you need to review?

**Review by ladyada — APPROVED** [review]

**Comment by ladyada** [comment]
lgtm thank you for checking!

File: ports/raspberrypi/boards/sparkfun_micromod_rp2040/pins.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
+#include "shared-bindings/board/__init__.h"
+
+STATIC const mp_rom_map_elem_t board_global_dict_table[] = {
+    // D (Digital only) pins (D0,D1)
+    { MP_ROM_QSTR(MP_QSTR_D0), MP_ROM_PTR(&pin_GPIO7) },        // GPIO6 - D0
```
Comment (zapwizard) [inline ports/raspberrypi/boards/sparkfun_micromod_rp2040/pins.c:5]: This calls out GPI07 twice (Comment looks like it should be GPIO6)

**Review by zapwizard — COMMENTED** [review]
Couldn't figure out why I can't talk to a SPI display, D0 looks to be mapped wrong

File: ports/raspberrypi/boards/sparkfun_micromod_rp2040/pins.c (hunk 1 above)
Comment (stonehippo) [reply ports/raspberrypi/boards/sparkfun_micromod_rp2040/pins.c:5]: @zapwizard Thanks for catching that. I've created #4665 to address it.

Not sure that should block connecting with SPI, though. Are you using the default definition? I'll take a look an make sure that's correct.

**Pull request merged**